rm(list = ls())
setwd("~/Dropbox/Wisconsin/Papers/GWAS/r2")

library(ggplot2)
library(dplyr)
library(stringr)
library(tidyverse)
library(broom)

source("LDFunctions.R")


# Generated by vcftools for the entire VCF file
# vcftools --vcf dean_converted.vcf --geno-r2 --ld-window 100
LD <- read.table(file = "r2-ldwindow_100", sep="\t", header=TRUE, stringsAsFactors = F)

# Split big matrix into list separated into chromosomes
LD.by.chroms <- splitByChrom(LD, nchrom = 9, sep = "_")

# Loop over chromosomes and get the distance between the marker pairs (retaining the corresponding r2)
LD.by.dist <- calcDist(LD.by.chroms)

# save(LD.by.dist, file = "r2Decay.RData")

modelFit <- fitDecay(LD.by.dist, 10)

png("firstMegabase.png", units="in", width=15, height=10, res=750)
modelFit$ggplot
dev.off()




