rm(list = ls())
setwd("~/Dropbox/Wisconsin/Papers/GWAS/r2")

library(ggplot2)
library(dplyr)
library(stringr)
library(tidyverse)
library(broom)
library(qqman)
source("LDFunctions.R")

# Generated by vcftools for the entire VCF file
# vcftools --vcf dean_converted.vcf --geno-r2 --ld-window 100


# Generated by vcftools for the entire VCF file
# vcftools --vcf dean_converted.vcf --geno-r2 --ld-window 100
LD <- read.table(file = "r2-ldwindow_100", sep="\t", header=TRUE, stringsAsFactors = F)

# Split big matrix into list separated into chromosomes
LD.by.chroms <- splitByChrom(LD, nchrom = 9, sep = "_")

# Loop over chromosomes and get the mean r2 in each window
LD.by.window <- slidingWindow(LD.by.chroms, windowSize = 100)


#save(LD.by.window, file = "r2SlidingWindow.RData")

allchroms <- LD.by.window[[10]]
allchroms <- allchroms[-c(which(is.na(allchroms$r2))),]

png("r2SlidingWindow-100SNPs.png", units="in", width=15, height=10, res=750)
manhattan(allchroms, chr = "chr", bp = "bp", p = "r2", logp = F, suggestiveline = F, genomewideline = F, ylab = expression("Linkage disequilbrium"~(r^{2})))
dev.off()



